M3/practica/clientes.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+if (empty ($_SERVER ['REQUEST_METHOD'])) {
+    header ('HTTP/1.1 401 Unauthorized');
+    exit();
+}
+
+$method = $_SERVER ['REQUEST_METHOD'];
+
+if (in_array ($method, array ('PUT', 'DELETE') )) {
+    $params = null;
+    parse_str (file_get_contents ('php://input'), $params);
+    // Add these request vars into _REQUEST, mimicing default behavior, PUT/DELETE will override existing COOKIE/GET vars
+    $_REQUEST = $params + $_REQUEST;
+}
+
+require_once ('cliente.php');
+
+if ($method == 'GET') {
+
+    if (empty ($_GET ['filtro']) || empty ($_GET ['desde']) || ! isset ($_GET ['cantidad']) ) {
+        if (empty ($_GET ['filtro']) || $_GET ['filtro'] == 'impar') {
+            sleep(3);
+        }
+        header ('HTTP/1.1 401 Unauthorized');
+        exit();
+    }
+
+    $filtro = $_GET ['filtro'];
+    $desde = $_GET ['desde'];
+    $cantidad = $_GET ['cantidad'];
+
+    if (! in_array ($filtro, array ('impar', 'par')) ||
+        ! is_numeric ($desde) || $desde < 1 ||
+        ! is_numeric ($cantidad) || $cantidad < 0
+    ) {
+        if ($filtro == 'impar') {
+            sleep(3);
+        }
+        header ('HTTP/1.1 401 Unauthorized');
+        exit();
+    }
+
+    if ($cantidad == 0) {
+        if ($filtro == 'impar') {
+            sleep(3);
+        }
+        echo '[]';
+        exit();
+    }
+
+    $db = @new mysqli ('localhost', 'root', '', 'cursoweb');
+    if ($db -> connect_error) {
+        if ($filtro == 'impar') {
+            sleep(3);
+        }
+        header ('HTTP/1.1 500 Internal Server Error');
+        exit();
+    }
+
+    /*
+    $query = 'SELECT COUNT(id) FROM clientes;';
+    $resultado = $db -> query ($query);
+    if ($db -> error) {
+        $db -> close();
+        if ($filtro == 'impar') {
+            sleep (3);
+        }
+        header ('HTTP/1.1 500 Internal Server Error');
+        exit();
+    }
+    
+    if ($cantidad > $resultado -> fetch_assoc() ['COUNT(id)']) {
+    
+    }
+    */
+
+    $paridad = $filtro == 'impar' ? 1 : 0;
+    $query = "SELECT id,
+                     razon_social,
+                     logo,
+                     direccion_calle,
+                     direccion_altura,
+                     direccion_ciudad,
+                     activo,
+                     pasivo,
+                     presidente_nombre,
+                     presidente_apellido
+              FROM clientes WHERE id >= {$desde} AND id % 2 = {$paridad} LIMIT {$cantidad} ;";
+    $resultado = $db -> query ($query);
+    if ($db -> error) {
+        $db -> close();
+        if ($filtro == 'impar') {
+            sleep(3);
+        }
+        header ('HTTP/1.1 500 Internal Server Error');
+        exit();
+    }
+
+    $db -> close();
```

Comment:
"Las conexiones de MySQL no persistentes y los conjuntos de resultados son automáticamente destruidos cuando un script de PHP finaliza su ejecución."
Warning: main(): Couldn't fetch mysqli in C:\Program Files\XAMPP\htdocs\cursoweb\M3\practica\clientes.php on line 11

Eso pasa si lo cerras y despues haces if ($db -> error)